contracts/optimistic-ethereum/OVM/predeploys/OVM_Upgrader.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+
+/* Library Imports */
+import { Lib_BytesUtils } from "../../libraries/utils/Lib_BytesUtils.sol";
+import { Lib_OVMCodec } from "../../libraries/codec/Lib_OVMCodec.sol";
+import { Lib_ECDSAUtils } from "../../libraries/utils/Lib_ECDSAUtils.sol";
+import { Lib_SafeExecutionManagerWrapper } from "../../libraries/wrappers/Lib_SafeExecutionManagerWrapper.sol";
+
+contract OVM_Upgrader {
+    // Account which can initiate upgrades on L1 by approving a hash onion.
+    address public l1UpgradeAuthorizer;
+
+    function doUpgrade(
+        address _address,
+        bytes memory _code
+    ) external {
```

Comment:
Not authenticated yet. I assume this needs to be authenticated for this to work.

Reply:
correct, will add a few comments

Reply:
updated to be more explicitly stubby